lib/hex_web/assets.ex

```diff
@@ -152,5 +152,5 @@ defmodule HexWeb.Assets do
 
   def tarball_cdn_key(release), do: "tarballs/#{release.package.name}-#{release.version}"
 
-  def tarball_store_key(release), do: "tarballs/#{release.package.name}-#{release.version}.tar"
+  def tarball_store_key(release), do: "tarballs/#{release.package.name}-#{release.version}.tar.gz"
```

Comment:
I think it wasn't noticed because the key only matters on release revert, which is very rare, and if you happen to connect to a fastly endpoint where the old tarball was already cached. 

Great find!